src/main/java/com/icthh/xm/tmf/ms/activation/service/RetryService.java

```diff
@@ -69,6 +72,7 @@ public void retry(SagaEvent sagaEvent, SagaTransaction sagaTransaction, SagaTask
             log.warn("Retry limit exceeded for event {}. {} > {}", sagaEvent, sagaEvent.getRetryNumber(),
                 sagaTaskSpec.getRetryCount());
             try {
+                self.changeTransactionState(FAILED, sagaTransaction);
```

Comment:
maybe better to move this code to method retryLimitExceeded, like default behavior that we can override by lep?

Reply:
the main goal of placing this code here - be insure that state will be "failed" before retry Limit Exceeded will fire

Reply:
but we have handler of retry exceeded `retryLimitExceededWithTaskTypeResolver->retryLimitExceededWithTransactionTypeResolver->retryLimitExceeded` where we can handle situation when retry limit exceeded, and i think it's correct place to mark transaction failed. 

If we mark transaction failed here, we can broke existing logic that do some intelligent retry or other operation that required transaction in state new.

my suggestion move this link inside method retryLimitExceeded

Reply:
this will broke some our logic, I will have situation that  in retryLimitExceededWithTaskTypeResolver transaction state is not commited do db

Reply:
changeTransactionState moved to changeTransactionState method